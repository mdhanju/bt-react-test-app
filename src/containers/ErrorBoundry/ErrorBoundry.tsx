import { FormattedMessage } from 'react-intl'
import React, { ReactNode } from 'react'

type Props = {
    children?: ReactNode;
};

type State = {
    error: Error | null;
    errorInfo: { componentStack?: string } | null;
};

class ErrorBoundary extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props)
        this.state = { error: null, errorInfo: null }
    }

    componentDidCatch(error: Error, errorInfo: object): void {
        this.setState({
            error: error,
            errorInfo: errorInfo
        })
    }

    render(): React.ReactNode {
        if (this.state.errorInfo) {
            return (
                <div>
                    <h2>
                        <FormattedMessage id="errorboundry.error" />
                    </h2>
                    <details style={{ whiteSpace: 'pre-wrap' }}>
                        {this.state.error && this.state.error.toString()}
                        <br />
                        {this.state.errorInfo.componentStack}
                    </details>
                </div>
            )
        }
        // Render children if there's no error
        return this.props.children
    }
}
export default ErrorBoundary
