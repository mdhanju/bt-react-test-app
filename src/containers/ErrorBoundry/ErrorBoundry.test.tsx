/* eslint-disable react/jsx-no-undef */

import React from 'react'
import { shallow, render, mount } from 'enzyme'
import ErrorBoundry from './ErrorBoundry'

const Buggy = (): never => {
    throw new Error('An error has occured in Buggy component!')
}

describe('<ErrorBoundry />', () => {
    it('should render properly without crashing', () => {
        const wrapper = shallow(<ErrorBoundry />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('render the ErrorBoundry with no errors', () => {
        const TestComp = (): React.ReactElement => {
            return <div>Test</div>
        }
        const wrapper = render(
            <ErrorBoundry>
                <TestComp />
            </ErrorBoundry>
        )
        expect(wrapper.html()).toBe('Test')
    })

    it('render the ErrorBoundry with errors', () => {
        sinon.spy(ErrorBoundry.prototype, 'componentDidCatch')
        mount(
            <Intl locale="en">
                <ErrorBoundry>
                    <Buggy />
                </ErrorBoundry>
            </Intl>
        )
        expect(ErrorBoundry.prototype.componentDidCatch).toHaveProperty(
            'callCount',
            1
        )
    })
})
