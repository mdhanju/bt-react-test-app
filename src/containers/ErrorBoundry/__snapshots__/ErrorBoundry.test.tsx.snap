// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundry /> should render properly without crashing 1`] = `
Array [
  null,
]
`;
